packages/connect-wallet-modal/src/components/WalletsModal/components/ConnectWalletModal/ConnectWalletModal.tsx

```diff
@@ -32,58 +33,76 @@ export const ConnectWalletModal = ({
   const [inputValue, setInputValue] = useState('');
   const [isShownOtherWallets, setShowOtherWallets] = useState(false);
 
-  const walletsListFull = useMemo(() => {
-    return sortWalletsList({
+  const handleInputChange = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => {
+      setInputValue(e.currentTarget.value);
+    },
+    [],
+  );
+
+  const handleInputClear = useCallback(() => {
+    setInputValue('');
+  }, []);
+
+  const handleToggleWalletsList = useCallback(() => {
+    setShowOtherWallets((value) => !value);
+  }, []);
+
+  const isWalletsToggleButtonShown = useRef(false);
+
+  const [walletsList, setWalletsList] = useState<WalletAdapterData[]>([]);
+  const [walletsListFull, setWalletsListFull] = useState<WalletAdapterData[]>(
+    [],
+  );
+
+  const getWalletsListFull = useCallback(async () => {
+    // Asynchronously filling wallets list because there is an async wallet detection during wallets sorting.
+    // Actually, almost all wallets can be detected synchronously, so this process is expected to be fast,
+    // and a loading indicator is not required for now.
+    const _walletsListFull = await sortWalletsList({
       walletDataList,
       walletsShown,
       walletsPinned,
     });
-  }, [walletDataList, walletsShown, walletsPinned]);
+    setWalletsListFull(_walletsListFull);
+  }, [walletDataList, walletsPinned, walletsShown]);
 
-  const walletsList = useMemo(() => {
+  useEffect(() => {
+    void getWalletsListFull();
+  }, [getWalletsListFull]);
+
+  useEffect(() => {
+    let _walletsList = walletsListFull;
     if (!isShownOtherWallets) {
-      return walletsListFull.slice(0, walletsDisplayInitialCount);
+      _walletsList = walletsListFull.slice(0, walletsDisplayInitialCount);
     }
 
     if (inputValue) {
-      return walletsListFull.filter((wallet) =>
+      _walletsList = walletsListFull.filter((wallet) =>
         wallet.walletName.toLowerCase().includes(inputValue.toLowerCase()),
       );
     }
 
-    return walletsListFull;
+    setWalletsList(_walletsList);
+    isWalletsToggleButtonShown.current =
+      walletsListFull.length > walletsList.length || isShownOtherWallets;
   }, [
     inputValue,
-    walletsListFull,
     isShownOtherWallets,
+    walletDataList,
     walletsDisplayInitialCount,
+    walletsList.length,
+    walletsListFull,
```

Comment:
these variables are no longer needed here. Please check deps

Reply:
> ESLint: React Hook useEffect has missing dependencies: 'walletsList.length' and 'walletsListFull'. Either include them or remove the dependency array.(react-hooks/exhaustive-deps)

These deps are generated by `eslint --fix`.

Reply:
I mean walletDataList, walletsPinned, walletsShown

Reply:
You are right, fixed